crypto/sm2/sm2_pmeth.c

```diff
@@ -255,6 +259,16 @@ static int pkey_sm2_ctrl_str(EVP_PKEY_CTX *ctx,
     } else if (strcmp(type, "sm2_id") == 0) {
         return pkey_sm2_ctrl(ctx, EVP_PKEY_CTRL_SET1_ID,
                              (int)strlen(value), (void *)value);
+    } else if (strcmp(type, "sm2_hex_id") == 0) {
```

Comment:
While intuitive, this will create a problem with a provider based SM2 implementation, because how `OSSL_PARAM_construct_from_text` / `OSSL_PARAM_allocate_from_text` treats this kind of string in an OSSL_PARAM context...  they expect a *prefix* "hex" (i.e. "hexsm2_id" in this case), because that's the pattern we've used so far.  Of course, it's perfectly possible to make those functions more "intelligent" and have them detect a "_hex_" infix, so I'm unsure how to think about this...

Note that this is a *side note*, so I'm not expecting you to change anything, we just need to be aware of this...  or well, I would appreciate it if you added this comment:

``` C
/*
 * TODO(3.0): reconsider the name, OR change OSSL_PARAM_construct_from_text() /
 * OSSL_PARAM_allocate_from_text() to handle infix "_hex_"
 */
```

Reply:
I prefer to make those OSSL_PARAM functions more intelligent since the name in the future might be various. So I am going to add the comment...

Reply:
(do note that since this hasn't been released yet, we can change our minds later, as long as we get to a final decision before releasing 3.0)

Reply:
Right, that is very reasonable